src/remark/code-block-snippets.js

```diff
@@ -0,0 +1,125 @@
+// @ts-check
+const fs = require('fs-extra');
+const path = require("path");
+const visit = require('unist-util-visit');
+const viewLinkUrl = require('../editUrl.js').viewLinkUrl;
+
+module.exports = function plugin() {
+    let filePath = ""
+
+    /**
+     * Searches for ![code](source) pattern and replaces it with the code block containing the
+     * content of the source file. It also supports region names inside the source file defined
+     * by #region and #endregion literals.
+     *
+     * Examples:
+     * ![code](../examples/my-example/shell-dump.txt)
+     * ![code go](../examples/my-example/src/main.go)
+     * ![code go](../examples/my-example/src/server.go "src/server.go")
+     * ![code go](../examples/my-example/src/server.go#L3)
+     * ![code go](../examples/my-example/src/server.go#L5-L11 "src/server.go")
+     * ![code go](../examples/my-example/src/account.go#native-account "Structure of the Native account")
+     */
+    function visitor(node) {
+        if (node.alt && node.alt.startsWith("code")) {
+            node.type = 'code';
+
+            // Check for language as the second argument. e.g. ![code rust](some-file.rs)
+            node.lang = node.alt.split(" ")[1];
+
+            // Use image title for the code block title.
+            if (node.title) {
+                node.meta = `title="${node.title}"`;
+            }
+
+            // Source filename can have anchor symbol # followed by the line numbers or region name.
+            const urlAnchor = node.url.split("#");
```

Comment:
I think there's too many similarly named variables, so:
```suggestion
            const [relativeSourceFilePath, anchor] = node.url.split("#");
```
I'd also rename "anchor" to fragment - https://developer.mozilla.org/en-US/docs/Web/API/URL/hash